pkg/mount/block/blockdev_linux.go

```diff
@@ -556,12 +556,60 @@ func (b BlockDevices) FilterPartLabel(label string) BlockDevices {
 	return b.FilterNames(names...)
 }
 
+// FilterAllowPCIString parses a string in the format vendor:device,vendor:device
+// and returns a list of BlockDev objects whose backing pci devices match
+// the vendor:device pairs passed in. All values are treated as hex.
+// E.g. 0x8086:0xABCD,8086:0x1234
+func (b BlockDevices) FilterAllowPCIString(allowlist string) (BlockDevices, error) {
+	pciList, err := parsePCIList(allowlist)
+	if err != nil {
+		return nil, err
+	}
+	return b.FilterAllowPCI(pciList), nil
+}
+
+// FilterAllowPCI returns a list of BlockDev objects whose backing
+// pci devices match the allowlist of PCI devices passed in.
+// FilterAllowPCI discards entries which don't have a matching PCI vendor
+// and device ID as an entry in the allowlist.
+func (b BlockDevices) FilterAllowPCI(allowlist pci.Devices) BlockDevices {
+	type mapKey struct {
+		vendor, device uint16
+	}
+	m := make(map[mapKey]bool)
+
+	for _, v := range allowlist {
+		m[mapKey{v.Vendor, v.Device}] = true
+	}
+	Debug("allow map is %v", m)
+
+	partitions := make(BlockDevices, 0)
+	for _, device := range b {
+		p, err := device.PCIInfo()
+		if err != nil {
+			// In the case of an error, we err on the safe side and choose not to block it.
+			// Not all block devices are backed by a pci device, for example SATA drives.
+			Debug("Failed to find PCI info; %v", err)
+			partitions = append(partitions, device)
```

Comment:
Do we actually need append in here ? 

in this case we can't confirm it is a PCI device so we can't confirm it's device and vendor ids match any of that in allow list.

Reply:
I believe we need remove it here given it is "allow" list now, 

For "block" list, if we cannot affirmatively confirm it matches block list, it chose not to block it, "on the safe" side, see the comment you preserved

In case of allowing, if we can't affirmatively confirm it matches allowlist, the safe side is ? (we are not sure if we should allow it) -- seems not allow it

Reply:
Ah yep I see what you're saying and that makes sense, I've removed it.